pkg/azurefile/nodeserver.go

```
@@ -160,63 +156,77 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
+	if isDiskMount {
+		if diskName == "" {
+			return nil, status.Errorf(codes.Internal, "diskname could not be empty, targetPath: %s", targetPath)
+		}
+		cifsMountFlags = []string{"dir_mode=0777,file_mode=0777,cache=strict,actimeo=30"}
+		cifsMountPath = filepath.Join(filepath.Dir(targetPath), proxyMount)
+	}
 
 	if runtime.GOOS == "windows" {
 		mountOptions = []string{fmt.Sprintf("AZURE\\%s", accountName), accountKey}
 	} else {
-		if err := os.MkdirAll(targetPath, 0700); err != nil {
+		if err := os.MkdirAll(targetPath, 0750); err != nil {
 			return nil, err
 		}
 		// parameters suggested by https://azure.microsoft.com/en-us/documentation/articles/storage-how-to-use-files-linux/
 		options := []string{fmt.Sprintf("username=%s,password=%s", accountName, accountKey)}
-		mountOptions = util.JoinMountOptions(mountFlags, options)
+		mountOptions = util.JoinMountOptions(cifsMountFlags, options)
 		mountOptions = appendDefaultMountOptions(mountOptions)
 	}
 
-	klog.V(2).Infof("target %v\nfstype %v\nvolumeId %v\ncontext %v\nmountflags %v\nmountOptions %v\n",
-		targetPath, fsType, volumeID, attrib, mountFlags, mountOptions)
+	if err := d.ensureMountPoint(cifsMountPath); err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not mount target %q: %v", cifsMountPath, err)
+	}
+	klog.V(2).Infof("cifsMountPath(%v) fstype(%v) volumeId(%v) context(%v) mountflags(%v) mountOptions(%v)",
+		cifsMountPath, fsType, volumeID, attrib, mountFlags, mountOptions)
 
 	mountComplete := false
 	err = wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
-		err := d.mounter.Mount(source, targetPath, "cifs", mountOptions)
+		err := d.mounter.Mount(source, cifsMountPath, cifs, mountOptions)
 		mountComplete = true
 		return true, err
 	})
 	if !mountComplete {
-		return nil, fmt.Errorf("volume(%s) mount on %s timeout(10m)", volumeID, targetPath)
+		return nil, fmt.Errorf("volume(%s) mount on %s timeout(10m)", volumeID, cifsMountPath)
```

should return the error code